src/Libraries/CoreNodeModels/Range.cs

```diff
@@ -36,6 +38,33 @@ public override bool IsConvertible
 
         public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
         {
+            if (IsPartiallyApplied)
+            {
+                var connectedPorts = Enumerable.Range(0, inputAstNodes.Count)
```

Comment:
is there any reason to use this expression rather than:                 
`var connectedPorts = Enumerable.Range(0, this.InPorts.Count).Where(this.HasInput).ToList();`
?